inference_experimental/inference_exp/models/auto_loaders/models_registry.py

```diff
@@ -12,7 +13,15 @@
 EMBEDDING_TASK = "embedding"
 
 
-REGISTERED_MODELS: Dict[Tuple[ModelArchitecture, TaskType, BackendType], LazyClass] = {
+@dataclass(frozen=True)
+class RegistryEntry:
```

Comment:
can we simplify by having it be just RegistryEntry with:
  - module_name
  - class_name
  - Optional[model_feature]
  
then below in the list contributors don't have to choose or wonder about difference between using LazyClass and RegistryEntry.  could instantiate LazyClass or resolve directly in  `resolve_model_class` if we don't do it elsewhere?